site/content/contribute/server/style-guide.md

```diff
@@ -0,0 +1,98 @@
+---
+title: "Style guide"
+date: 2021-01-12T16:00:00+0530
+weight: 3
+subsection: Server
+---
+
+The overarching guideline is to just follow [Effective Go](https://golang.org/doc/effective_go.html) along with [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments). But currently, implementing those rules can lead to a clash between maintaining consistency and enforcing correctness. This document aims to codify some of the rules in the hope that any confusion is alleviated.
+
+### Model package
+
+The `model` package is our public API which is consumed by multiple plugins. Having a non-idiomatic public API should not prevent us from improving the rest of the codebase. More code keeps getting added, and it becomes practically impossible to fix all of the issues at once. Instead, we choose to have consistency in that package only if changing it makes it inconsistent from an API perspective. This includes both breaking changes and any inconsistencies that might appear in godocs.
```

Comment:
```suggestion
The `model` package is our public API which is consumed by many plugins and 3rd party integrations. The need to maintain backwards-compatibility for these external users of the `model` package prevents us from immediately bringing it into compliance with some of the rules outlined in this document.

In order to avoid delaying the adoption of these rules in the wider codebase, we have chosen to temporarily exempt only the `model` package from certain rules (the affected rules are indicated below). However, outside of the `model` package, these rules should be followed in all new or modified code.
```